src/app/static/src/app/store/baseline/actions.ts

```diff
@@ -19,24 +19,23 @@ export interface BaselineActions {
 
 export const actions: ActionTree<BaselineState, RootState> & BaselineActions = {
 
-    uploadPJNZ({commit}, file) {
+    async uploadPJNZ({commit}, file) {
         let formData = new FormData();
         formData.append('file', file);
-        api.postAndReturn<PjnzResponse>("/baseline/pjnz/", formData)
-            .then((payload) => {
-                commit<BaselinePayload<PjnzResponse>>({type: "PJNZUploaded", payload});
-            })
-            .catch((error: Error) => {
-                commit<BaselinePayload<String>>({type: 'PJNZUploadError', payload: error.message});
-            });
+
+        const payload = await api()
+            .commitFirstErrorAsType(commit, "PJNZUploadError")
```

Comment:
Just occurred to me that we could extend this fluent interface further so we have something like:
```
await api(commit).commitFirstErrorAs("PJNZUploadError")
                .commitSuccessAs<PjnzResponse>("PJNZUploaded")
                .post("/baseline/pjnz/", formData)
```

Reply:
I think this would be good, to keep this all in one statement, it feels a little disjointed to have it in two. This would be really terse and clean. 

What I do prefer about the old way of doing things is the ordering:
1. what to do
2. what to do if 1. succeeds
3. what to do if 1. fails

Having the primary action come last in the statement feels a little bit backwards, and less readable. But I guess I could get used to it! Not sure how you could achieve it the other way round anyway, you'd need the initial 'post' to not actually do the post, but wait for the other commit setups. 
```
api(commit)
  .post(..)
  .commitSuccessAs(..)
  .commitFirstErrorAs(..)
  .go() 
``` 
would hardly be better!